runtime/Java/src/org/antlr/v4/runtime/atn/ATNDataReader.java

```diff
@@ -0,0 +1,42 @@
+package org.antlr.v4.runtime.atn;
+
+import java.util.UUID;
+
+public abstract class ATNDataReader {
+	protected abstract byte readByte();
+
+	public int read() {
```

Comment:
Is there a standard base64 decoder somewhere?

Reply:
Oh i see base64 below now. Hmm...adding general comment outside.

Reply:
It's the binary reader that uses abstract `readByte` calls. `readByte` is implemented in `ATNDataReaderBase64` (that reads from base64 string) and in `ATNDataReaderByteBuffer`.